app/assets/javascripts/3dbio_viewer/src/webapp/components/RootViewerContents.tsx

```diff
@@ -158,6 +150,72 @@ function getUploadData(externalData: ExternalData) {
         : undefined;
 }
 
+function useResizableBox() {
+    const [innerWidth, setInnerWidth] = React.useState(window.innerWidth);
+
+    const resizableBoxProps = React.useMemo<
+        Required<
+            Pick<ResizableBoxProps, "width" | "minConstraints" | "maxConstraints" | "resizeHandles">
+        >
+    >(
+        () => ({
+            width: innerWidth * 0.55,
+            minConstraints: [400, 0],
+            maxConstraints: [innerWidth - 600, 0],
+            resizeHandles: ["w"],
+        }),
+        [innerWidth]
+    );
+
+    React.useLayoutEffect(() => {
+        function updateInnerWidth() {
+            setInnerWidth(window.innerWidth);
+        }
+        window.addEventListener("resize", updateInnerWidth);
+        return () => window.removeEventListener("resize", updateInnerWidth);
+    }, []);
+
+    return { innerWidth, resizableBoxProps };
+}
+
+function useGoToTop<K extends HTMLElement>() {
+    const [scrolled, setScrolled] = React.useState(false);
+    const ref = React.useRef<K>(null);
+
+    const goToTop = React.useCallback(() => {
+        if (!ref.current) return;
+        ref.current.scrollTop = 0;
+    }, [ref]);
+
+    React.useEffect(() => {
+        function scrollFunction() {
+            if (ref.current) setScrolled(ref.current.scrollTop > 20);
+        }
+        const el = ref.current; //suggested by eslint
+        if (el) {
+            el.addEventListener("scroll", scrollFunction);
+        }
+        return () => {
+            if (el) el.removeEventListener("scroll", scrollFunction);
+        };
+    }, []);
```

Comment:
Not serious, but I have a doubt about the block. It's a useEffect without dependencies, so it runs on component mount, right? The thing is, `ref.current` may not exist, sono addEventListener. So it could be a check on the top of the block. But then, how are we sure that the listener is always added?